libqtile/group.py

```diff
@@ -211,6 +238,9 @@ def focus(self, win, warp=True, force=False):
         if self.qtile._drag and not force:
             # don't change focus while dragging windows (unless forced)
             return
+        if FocusLock.is_active():
```

Comment:
I guess rather than a global lock, I would prefer something like (untested):

```
for _, w in self.wigets_map.items():
    if isinstance(w, widgets.Prompt) and w.active:
        return
```

largely because I don't think we have other widgets that steal focus like prompt does, so I'm not sure how useful this is outside of that.

Reply:
> (can_steal_focus() would then have to hold the loop with the prompt widget you mentioned.)

I think that's fine; I have three monitors and only 14 widgets total across all three. I guess some very fancy configs might have 3-5x that, but still much less than 100. If we're making focus changes, I expect the roundtrips for doing the map, updating _NET_WM_ACTIVE_WINDOW, etc. to dwarf the cost of this loop. The only case where we wouldn't pay that cost is if we're detecting a prompt is active, which is a rare event.

Reply:
Kk, I'll look into that route then.